common/clickteam.cpp

```diff
@@ -0,0 +1,306 @@
+	delete[] stub;
+	return true;
+}
+
+struct TagHead {
+	uint16 id;
+	uint16 flags;
+	uint32 compressedLen;
+};
+
+ClickteamInstaller* ClickteamInstaller::open(Common::SeekableReadStream *stream, DisposeAfterUse::Flag dispose)
+{
+	Common::HashMap<Common::String, ClickteamFileDescriptor> files;
+	Common::HashMap<Common::String, Common::String> caseMap;
+	HashMap<uint16, Common::SharedPtr<ClickteamTag>> tags;
+	uint32 crc_xor;
+
+	if (!checkStubAndComputeCRC1(stream, crc_xor))
+		return nullptr;
+
+	int64 block3_offset = -1, block3_len = 0;
+
+	while (!stream->eos()) {
+		TagHead tagHead;
+		stream->read(&tagHead, sizeof(tagHead));
+		uint16 tagId = FROM_LE_16(tagHead.id);
+		uint16 flags = FROM_LE_16(tagHead.flags);
+		uint32 compressedPayloadLen = FROM_LE_32(tagHead.compressedLen);
+		if (tagId == 0x7f7f) {
+			stream->skip(4);
+			block3_offset = stream->pos();
+			block3_len = stream->readUint32LE();
+			break;
+		}
+		if (compressedPayloadLen == 0) {
+			break;
+		}
+		byte *compressedPayload = new byte[compressedPayloadLen];
+		stream->read(compressedPayload, compressedPayloadLen);
+
+		ClickteamTag *tag;
+
+		if (flags & FLAG_COMPRESSED) {
+			if (compressedPayloadLen < 4) {
+				delete[] compressedPayload;
+				continue;
+			}
+			uint32 uncompressedPayloadLen = READ_LE_UINT32(compressedPayload);
+			byte *uncompressedPayload = new byte[uncompressedPayloadLen];
+			int32 ret = Common::GzioReadStream::clickteamDecompress(uncompressedPayload,
+										  uncompressedPayloadLen,
+										  compressedPayload + 4,
+										  compressedPayloadLen - 4);
+			delete[] compressedPayload;
+			if (ret < 0) {
+				debug ("Decompression error");
+				continue;
+			}
+			tag = new ClickteamTag(tagId, uncompressedPayload, uncompressedPayloadLen);
+		} else {
+			tag = new ClickteamTag(tagId, compressedPayload, compressedPayloadLen);
+		}
+		tags[tagId].reset(tag);
+		switch (tag->_tagId) {
+		case (uint16) ClickteamTagId::FILE_LIST: {
+			if (tag->_size < 4) {
+				return nullptr;
+			}
+			uint32 count = READ_LE_UINT32(tag->_contents);
+			uint32 off = 4;
+			for (unsigned i = 0; i < count && off + 0x24 < tag->_size; i++) {
+				uint16 l = READ_LE_UINT16(tag->_contents + off);
+				if (l < 0x24)
+					break;
+				ClickteamFileDescriptor desc(*tag, off);
+				files[desc._fileName] = desc;
+				Common::String lower = desc._fileName;
+				lower.toLowercase();
+				caseMap[lower] = desc._fileName;
+				off += l;
+			}
+			break;
+		}
+		case 0x1237: {
+			byte *p;
+			for (p = tag->_contents; p < tag->_contents + tag->_size; p++)
+				if (!*p)
+					break;
+			crc_xor = computeCRC(tag->_contents, p - tag->_contents, crc_xor);
+			break;
+		}
+		}
+	}
+
+	if (block3_offset <= 0 || block3_len <= 0)
+		return nullptr;
+
+	return new ClickteamInstaller(files, caseMap, tags, crc_xor, block3_offset, block3_len, stream, dispose);
+}
+
+static Common::String translateName(const Path &path) {
+	Common::String lower = path.toString('\\');
```

Comment:
You could use '/' as a separator and then call `Common::normalizePath()` which is doing exactly the same thing.

Reply:
Done while using sep = '\\' argument to normalizePath. Archive uses \ internally, so it's simpler without translation